src/components/GroupedTaskStatusBadge/index.tsx

```diff
@@ -1,45 +1,35 @@
 import styled from "@emotion/styled";
 import Tooltip from "@leafygreen-ui/tooltip";
-import { Link, useLocation } from "react-router-dom";
+import { Link } from "react-router-dom";
 import { TaskStatusIcon } from "components/TaskStatusIcon";
-import { getVersionRoute } from "constants/routes";
+import { GetVersionRouteOptions, getVersionRoute } from "constants/routes";
 import {
   taskStatusToCopy,
   mapUmbrellaStatusColors,
   mapUmbrellaStatusToQueryParam,
 } from "constants/task";
 import { fontSize, size, zIndex } from "constants/tokens";
 import { TaskStatus } from "types/task";
-import { queryString, string } from "utils";
 
-const { applyStrictRegex } = string;
-const { parseQueryString } = queryString;
-
-interface Props {
-  status: TaskStatus;
+interface GroupedTaskStatusBadgeProps {
   count: number;
   onClick?: () => void;
+  status: TaskStatus;
   statusCounts?: { [key: string]: number };
   versionId: string;
-  variant?: string;
-  preserveSorts?: boolean;
+  queryParamsToPreserve?: GetVersionRouteOptions;
```

Comment:
Is this still necessary now that we no longer link from the side panel?

Reply:
I kind of like this pattern better, since it's more flexible 😅 I originally wrote the `preserveSorts` stuff and I don't really like it